pkg/edgehub/clients/wsclient/websocket_test.go

```diff
@@ -0,0 +1,405 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package wsclient
+
+import (
+	"net/http"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/pkg/common/util"
+	"github.com/satori/go.uuid"
+
+	"net/http/httptest"
+)
+
+//init() starts the test server and generates test certificates for testing
+func init() {
+	newTestServer()
+
+	err := util.GenerateTestCertificate("/tmp/", "edge", "edge")
+	if err != nil {
+		panic("Error in creating fake certificates")
+	}
+}
+
+//Message is content object to be passed in as message object
+type Message struct {
+	Name string `json:"name"`
+}
+
+var testServer *httptest.Server
+var upgrader = websocket.Upgrader{}
+
+//newTestServer() starts a fake server for testing
+func newTestServer() {
```

Comment:
it's better if we move this method to a websocket helper file . if any other tests need to start a fake web socket server. this method will be useful . @pavan187 , please guide.

Reply:
@rohitsardesai83  this function starting the websocker server on some hard coded api's, which can be used only to this test file, there are no standard/common api's this method is using, i feel other test functions cannnot reuse this method for websocker server testing.

Reply:
ok.